OneCommunityGlobal/HGNRest
# Shengwei fix user profile quick setup

# Description
<img width="538" alt="image" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/69882989/1fb27f14-e607-4332-8af3-f7b01b12f42a">

## Related PRS (if any):
This backend PR is related to the #[2339](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/2339) frontend PR.
…

## Main changes explained:
- **Integrate new getAllTeamCode API:** Leaner version of the teamcontroller.getAllTeams, returning active team code info

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. Clear site data/cache
4. log as admin user
5. go to user profile -> use the quick setup modal on the left side 
6. Verify that no duplicate project/team is added to the user profile with the use of quicket setup modal.
  - <img width="500" alt="image" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/69882989/1d1cd5b2-79ea-4269-adab-638e5108e789">
  - <img width="405" alt="image" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/69882989/f4cd91b1-3404-4c28-9f55-26473d587ae3">

7. Verify that after deleting all project/team data, using the existing quick setup configuration and applying changes will not lead to a page crash.
8. Verify that data validation of  'Add A New TItlte' form is working. E.g. Team code, team name, and project validation should work as expected.
   -  <img width="500" alt="image" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/69882989/ca8a975c-dc95-409c-b502-cd91d7a4478d">

9. Verify that 'Add A New Title' is working as expected by saving a new quick setup config.
10. Verify that after saving new info with use of quick setup modal, no warning message of 'please click on save changes…' displayed 
## Screenshots or videos of changes:

## Note:
Include the information the reviewers need to know.
